File: Day1/Day1.py
import re

def findNumber(line):
    line = re.sub('\\D', '', line)
    numbers = list(line)
    return int(numbers[0] + numbers[len(numbers) - 1])

file = open('Day1.txt', 'r')

countFirst = 0
countSecond = 0

for line in file.readlines():
    line = line.strip()

    abc = {
        1 : 'one',
        2 : 'two',
        3 : 'three',
        4 : 'four',
        5 : 'five',
        6 : 'six',
        7 : 'seven',
        8 : 'eight',
        9 : 'nine',
    }

    countFirst += findNumber(line)

    for key, val in abc.items():
        line = re.sub(val, val+str(key)+val, line)
    countSecond += findNumber(line)

print(countFirst)
print(countSecond)

File: Day1/Day1.php
<?php

function findNumber($line)
{
    $line = preg_replace('/\D/', '', $line);
    $numbers = str_split($line);
    return intval($numbers[0] . $numbers[count($numbers) - 1]);
}

$file = file('Day1.txt', FILE_IGNORE_NEW_LINES);

$countFirst = 0;
$countSecond = 0;

foreach ($file as $line) {
    $abc = [
        'one',
        'two',
        'three',
        'four',
        'five',
        'six',
        'seven',
        'eight',
        'nine',
    ];

    $countFirst += findNumber($line);
    foreach ($abc as $key => $val) {
        $line = str_replace($val, $val.strval($key + 1).$val, $line);
    }
    $countSecond += findNumber($line);
}

var_dump($countFirst);
var_dump($countSecond);